tests/drivers/gpio/gpio_basic_api/testcase.yaml

```diff
@@ -1,5 +1,36 @@
 tests:
   drivers.gpio.2pin:
+    platform_exclude: nucleo_f091rc nucleo_l476rg nucleo_l073rz nucleo_f401re stm32mp157c_dk2 nucleo_f030r8 nucleo_g071rb nucleo_l053r8 nucleo_g474re nucleo_f334r8 nucleo_wb55rg rv32m1_vega nucleo_f746zg mimxrt1050_evk frdm_k64f disco_l475_iot1 nrf52840dk_nrf52840 nucleo_l4r5zi lpcxpresso55s69 nucleo_f429zi mimxrt1060_evk hifive1_revb nucleo_l152re
```

Comment:
IMO this is unmaintainable, but we can discuss it in dev-review.

Reply:
compared to 20 duplicated files ?
I think I prefer adding a single board name here, instead of duplicating identical files everywhere.

Reply:
of course it might be possible to adjust testcase.yml scheme, so that if a platform in specifically mentioned further down, then it is automatically exclude here, but that would be an extension, but probably a valuable one.

Note, the reason I didn't line break, is because I'm unsure if the testcase yaml shema allows it, and / or if a proper yml sequence is allowed. So this was a quick way, to show the principle.

Reply:
The problem is, instead of inheriting support for all existing tests that use fixtures by simply adding a fixtures include to the board, you have to edit every test that supports a specific fixture and add the board into the appropriate build option, trying not to corrupt the existing list when you do so. 

Also the board-specific overlays providing support for fixtures aren't in the boards directory where people can find them, they're down under tests/common, are separate for each fixture, and do not indicate in the overlay what boards they apply to.

None of that is insurmountable, but I don't consider it an improvement over the current content of #30594.

Reply:
update.
It is possible to make new lines, which help on the long lines, but doesn't solve the issue with having to specify the board both in platform_exclude, and then a platform_allow.

But would like your opinion on this, if there can be an `platform_allow_exclude_others` kind of entry.